astore/rpc/auth.proto

```diff
@@ -15,15 +15,15 @@ message AuthenticateResponse {
 
 message TokenRequest {
   string url = 1; // URL returned by server.
+  bytes publickey = 2; // Public key to be signed by the server. Optional.
 }
 
 message TokenResponse {
   bytes nonce = 1; // Nonce used for encryption.
   bytes token = 2; // Encrypted token. Requires the private key corresponding to the public key supplied to open.
-  bytes key = 3; // Private Key used to login
-  bytes cert = 4; // Certificate signed to be used with the Private Key
-  bytes capublickey = 5; // CA Public Key to be added to the authenticated client
-  repeated string cahosts = 6; // List of hosts the CA should be trusted for
+  bytes cert = 3; // Certificate signed to be used with the Private Key, is a signed version of the public key sent in the TokenRequest
```

Comment:
I believe it does not matter in this instance, but generally you never re-number fields, that way protocol buffer maintains wire compatibility, addressing some compatibility / versioning issues.

Reply:
you know what, I did this knowing it wouldn't change anything, but it's the principal of the matter. I reverted the numbering.